pandas/tests/arrays/test_datetimelike.py

```diff
@@ -812,3 +812,13 @@ def test_to_numpy_extra(array):
     assert result[0] == result[1]
 
     tm.assert_equal(array, original)
+
+
+@pytest.mark.parametrize("klass", [list, np.array, pd.array, pd.Series])
+def test_searchsorted_datetimelike_with_listlike(klass):
+    # https://github.com/pandas-dev/pandas/issues/32762
+    dates = pd.to_datetime(["2020-01-01", "2020-02-01"])
+    result = dates.searchsorted(klass(dates))
+    expected = np.array([0, 1], dtype=result.dtype)
+
+    tm.assert_numpy_array_equal(result, expected)
```

Comment:
do we have sufficient tests for if we have non-matching dtypes e.g. integers?

what about different timezones, or tzawareness-compat?  (or periodarray freq mismatch)

Reply:
Added some tests for type mismatches